File: Automotive.Marketplace.Application/Features/VariantFeatures/CreateVariant/CreateVariantCommand.cs
using MediatR;

namespace Automotive.Marketplace.Application.Features.VariantFeatures.CreateVariant;

public sealed record CreateVariantCommand(
    Guid ModelId,
    Guid FuelId,
    Guid TransmissionId,
    Guid BodyTypeId,
    bool IsCustom,
    int DoorCount,
    int PowerKw,
    int EngineSizeMl
) : IRequest<CreateVariantResponse>;

File: automotive.marketplace.client/src/features/variantList/types/CreateVariantCommand.ts
export type CreateVariantCommand = {
  modelId: string;
  year: number;
  fuelId: string;
  transmissionId: string;
  bodyTypeId: string;
  isCustom: boolean;
  doorCount: number;
  powerKw: number;
  engineSizeMl: number;
};
